src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -198,6 +199,13 @@ static bool IsValidKey(byte[] key)
             using (var rijndael = new RijndaelManaged())
             {
                 var bitLength = key.Length * 8;
+
+                var maxValidKeyBitLength = rijndael.LegalKeySizes.OrderByDescending(keyLength => keyLength.MaxSize).Select(keyLength => keyLength.MaxSize).First();
+                if (bitLength < maxValidKeyBitLength)
+                {
+                    Log.WarnFormat("Encryption key is is {0} bits which is less than the maximum allowed {1} bits. Increasing the key length to {1} bits would provide stronger security.", bitLength, maxValidKeyBitLength);
```

Comment:
I would also re-phrase the second sentence to something like: "Consider using a {1}-bit encryption key to obtain the maximum cipher strength".

Security is multi-faceted and increasing algorithm strength might not automatically provide users with stronger protection.